app/controllers/tutorials_controller.rb

```diff
@@ -84,6 +95,23 @@ def bulk_download
     end
   end
 
+  def bulk_download_corrections
+    @zipped_corrections = Submission.zip_corrections!(@tutorial, @assignment)
+    if @zipped_corrections.is_a?(StringIO)
+    send_data @zipped_corrections.read,
+              filename: @assignment.title + '@' + @tutorial.title + '-Corrections.zip',
+              type: 'application/zip',
+              disposition: 'attachment'
+    else
+      flash[:alert] = I18n.t('controllers.tutorials.bulk_download_failed',
+                             message: @zipped_corrections)
+      redirect_to lecture_tutorials_path(@tutorial.lecture,
```

Comment:
Is it intended that the redirect is to `lecture_tutorials_path`? `lecture_tutorials_path` goes to 'tutorials#index' and there access is only granted to an actual tutor due to `before_action :check_tutot_status, only: :index`. So, this would lead to a no access warning for editors.